src/medal_widget/events.cljs

```diff
@@ -9,3 +9,18 @@
  ::initialize-db
  (fn-traced [_ _]
    db/default-db))
+
+(re-frame/reg-event-db
+ ::sort-it
+ (fn [db [_ item]]
+   (let [current-data (:medals-data db)
+         sorted (if (= item "gold")
+                  (sort #(or (> (get %1 (keyword item)) (get %2 (keyword item)))
+                             (and (= (get %1 (keyword item)) (get %2 (keyword item)))
+                                  (> (get %1 :total) (get %2 :total)))) current-data)
+                  (sort #(or (> (get %1 (keyword item)) (get %2 (keyword item)))
+                             (and (= (get %1 (keyword item)) (get %2 (keyword item)))
+                                  (> (get %1 :gold) (get %2 :gold)))) current-data))]
```

Comment:
This should be done in a subscription. As much as possible we should store the minimal state we need in the DB and derive everything else. So, what we should store in the db is the `:selected-sort-option` and then the subscription can use that along with the `:medals-data` to return the sorted list